contracts/StablecoinConverter.sol

```diff
@@ -40,25 +43,52 @@ contract StablecoinConverter is EpochTokenLocker {
         uint128 remainingAmount; // remainingAmount can either be a sellAmount or buyAmount, depending on the flag isSellOrder
     }
 
+    struct PreviousSolutionData {
+        uint32 batchId;
+        TradeData[] trades;
+        uint16[] tokenIdsForPrice;
+        address solutionSubmitter;
+        uint256 feeReward;
+        uint256 objectiveValue;
+    }
+
+    struct TradeData {
+        address owner;
+        uint128 volume;
+        uint16 orderId;
+    }
+
     // User-> Order
     mapping(address => Order[]) public orders;
 
+    // tokenId -> CurrentPrice
+    mapping (uint16 => uint128) public currentPrices;
+    PreviousSolutionData public previousSolution;
+
     // Iterable set of all users, required to collect auction information
     IterableAppendOnlySet.Data private allUsers;
     using IterableAppendOnlySet for IterableAppendOnlySet.Data;
 
     IdToAddressBiMap.Data private registeredTokens;
 
     uint public MAX_TOKENS;  // solhint-disable var-name-mixedcase
-    uint16 public numTokens = 0;
-    uint128 public feeDenominator; // fee is (1 / feeDenominator)
-
+    uint16 public numTokens;
+    uint128 public feeDenominator;
+
+    /** @dev Constructor determines exchange parameters
+      * @param maxTokens The maximum number of tokens that can be listed.
+      * @param _feeDenominator fee as a proportion is (1 / feeDenominator)
+      * @param feeToken Address of ERC20 fee token.
+      */
     constructor(uint maxTokens, uint128 _feeDenominator, address feeToken) public {
         MAX_TOKENS = maxTokens;
         feeDenominator = _feeDenominator;
         addToken(feeToken); // fee Token will always have the token index 0
     }
 
+    /** @dev Used to list a new token on the exchange.
+      * @param _tokenAddress ERC20 token to be listed.
```

Comment:
Maybe you could describe what listed means on our exchange

Reply:
@josojo - How does this look?

```
Used to list a new token on the contract: Hence, making it available for exchange in an auction.
```